add current type mutation workaround doc (#650)
timholy/Revise.jl: docs/src/limitations.md

- conflicts between variables and functions sharing the same name
- removal of `export`s

These kinds of changes require that you restart your Julia session. 

For type changes, you can use the workaround of renaming types with version number like:

```jl
# 1st version
struct FooStruct1
    bar::Int
end
FooStruct=FooStruct1
function processFoo(foo::FooStruct)
    @info foo.bar
end
```
and update type like
```jl
# 2nd version
struct FooStruct2  # change version here
    bar::Int
    str::String
end
FooStruct=FooStruct2   # change version here
function processFoo(foo::FooStruct)  # no need changing this
    @info foo.bar
end
```

> Then it could be used properly as long as the new type name don' t conflict with existing type name. So you can restart from version number 1 if you restart the REPL.

Reviewer: ```suggestion
This works as long as the new type name doesn' t conflict with an existing name; within a session you need to change the name each time you change the definition.

Once your development has converged on a solution, it's best to switch to the "permanent" name: in the example above, `FooStruct` is a non-constant global variable, and if used internally in a function there will be consequent performance penalties. Switching to the permanent name will force you to restart your session.
```

- conflicts between variables and functions sharing the same name
- removal of `export`s

> These kinds of changes require that you restart your Julia session. 

Reviewer 2: ```suggestion
These kinds of changes require that you restart your Julia session.
```

During early stages of development, it's quite common to want to change type definitions. You can work around Julia's/Revise's limitations by temporary renaming:

> ```jl

Reviewer 2: ````suggestion
```julia
````

# 1st version
struct FooStruct1
    bar::Int
end

> FooStruct=FooStruct1

Reviewer 2: ```suggestion
FooStruct = FooStruct1
```

function processFoo(foo::FooStruct)
    @info foo.bar
end
```
and then the type can be updated like

> ```jl

Reviewer 2: ````suggestion
```julia
````

# 2nd version
struct FooStruct2  # change version here
    bar::Int
    str::String
end

> FooStruct=FooStruct2   # change version here

Reviewer 2: ```suggestion
FooStruct = FooStruct2   # change version here
```

These kinds of changes require that you restart your Julia session.

During early stages of development, it's quite common to want to change type definitions. You can work around Julia's/Revise's limitations by temporary renaming:

```julia
# 1st version
struct FooStruct1
    bar::Int
end
FooStruct = FooStruct1
function processFoo(foo::FooStruct)
    @info foo.bar
end
```
and then the type can be updated like
```julia
# 2nd version
struct FooStruct2  # change version here
    bar::Int
    str::String
end
FooStruct = FooStruct2   # change version here
function processFoo(foo::FooStruct)  # no need to change this
    @info foo.bar
end
```

> This works as long as the new type name doesn' t conflict with an existing name; within a session you need to change the name each time you change the definition.

Reviewer: ```suggestion
This works as long as the new type name doesn't conflict with an existing name; within a session you need to change the name each time you change the definition.
```
